tests/testthat/test-module_teal.R

```diff
@@ -2974,3 +3034,118 @@ testthat::describe("teal-reporter", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = within(
+          teal.data::teal_data(),
+          iris <- iris
+        ),
+        modules = modules(
+          module("module_1", server = function(id, data) NULL)
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_module_id` = "module_1")
+        session$flushReact()
+        testthat::expect_match(output[["teal_modules-nav-module_1-source_code_wrapper-source_code_reason"]]$html,
+          "No source code is available for this module.",
+          fixed = TRUE
+        )
+      }
+    )
+  })
+
+  it("Show R code button contains 'is disabled' reason when disabled source code", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = within(
+          teal.data::teal_data(),
+          iris <- iris
+        ),
+        modules = modules(
+          module("module_1", server = function(id, data) data) |> disable_src()
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_module_id` = "module_1")
+        session$flushReact()
+        testthat::expect_match(output[["teal_modules-nav-module_1-source_code_wrapper-source_code_reason"]]$html,
+          "The source code functionality is disabled for this module.",
+          fixed = TRUE
+        )
+      }
+    )
+  })
+
+  it("Show R code button contains 'error' reason when module's server returns an error", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = within(
+          teal.data::teal_data(),
+          iris <- iris
+        ),
+        modules = modules(
+          module("module_1", server = function(id, data) reactive(teal.code::qenv(stop("test"))))
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_module_id` = "module_1")
+        session$flushReact()
+        testthat::expect_match(output[["teal_modules-nav-module_1-source_code_wrapper-source_code_reason"]]$html,
+          "The module returned an error, check it for errors.",
+          fixed = TRUE
+        )
+      }
+    )
+  })
+
+  it("Show R code button contains 'not support source code' reason when there is no code", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = teal.data::teal_data(),
+        modules = modules(
+          module("module_1", server = function(id, data) reactive(teal.code::qenv()))
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_module_id` = "module_1")
+        session$flushReact()
+        testthat::expect_match(output[["teal_modules-nav-module_1-source_code_wrapper-source_code_reason"]]$html,
+          "The module does not support source code functionality.",
+          fixed = TRUE
+        )
+      }
+    )
+  })
+  it("Show R code button doesn't contain reason when there is working code", {
```

Comment:
better to start `describe` with this - so it sets reader's understanding "what is expected" instead of "what is not expected, read everything to find out"